solr/core/src/java/org/apache/solr/response/transform/DocTransformer.java

```diff
@@ -41,12 +41,26 @@
    */
   public abstract String getName();
 
+
+  /**
+   * This is called before {@link #transform} to provide context for any subsequent transformations.
+   *
+   * @param context The {@link ResultContext} stores information about how the documents were produced.
+   * @see #needsSolrIndexSearcher
+   */
+  public void prepare( ResultContext context ) {
```

Comment:
if you want to do something different than setting the context (see LTRFeatureLoggerTransformerFactory for example), the name is misleading. we could keep the setContext and just add the `prepare` method.